system/solaris_zone.py

```diff
@@ -0,0 +1,446 @@
+#!/usr/bin/python
+
+# (c) 2015, Paul Markham <pmarkham@netrefinery.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+import sys
+import os
+import platform
+import tempfile
+
+DOCUMENTATION = '''
+---
+module: solaris_zone
+short_description: Manage Solaris zones
+description:
+   - Create, start, stop and delete Solaris zones. This module doesn't currently allow
+     changing of options for a zone that's already been created.
+version_added: "2.0"
+author: Paul Markham
+requirements:
+  - Solaris 10
+options:
+  state:
+    required: true
+    description:
+      - C(present), configure and install the zone.
+      - C(installed), synonym for C(present).
+      - C(running), if the zone already exists, boot it, otherwise, configure and install
+          the zone first, then boot it.
+      - C(started), synonym for C(running).
+      - C(stopped), shutdown a zone.
+      - C(absent), destroy the zone.
+      - C(configured), configure the ready so that it's to be attached.
```

Comment:
They're slightly different things.

There's three stages to setting up a zone. What you'd normally do is:

* configure the zone - defines the zone to the system -- it's not usable at this stage.
* install the zone - copy in all the operating system files the zone needs to run.
* boot the zone

The installed/present state does the first two; the running/started state does all three.

Attaching and detaching zones lets you moved them between physical servers.
When attaching a zone it's:

* configure the zone - optional if the zone has already been configured.
* attach the zone - point to existing file system with the operating system files.
* boot the zone

So we do need to differentiate between the states configured, installed and running.